packages/slice-machine/server/src/api/custom-types/delete.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  CustomTypesPaths,
+  GeneratedCustomTypesPaths,
+  MocksConfig,
+} from "../../../../lib/models/paths";
+import * as IO from "../../../../lib/io";
+import { getBackendState } from "../state";
+import { RequestWithEnv } from "../http/common";
+import { DeleteCustomTypeQuery } from "../../../../lib/models/common/CustomType";
+import { remove as removeCtsFromMockConfig } from "../../../../lib/mock/misc/fs";
+import path, { resolve } from "path";
+
+export default async function handler(
+  req: RequestWithEnv
+): Promise<
+  { err: unknown; reason: string; status: string } | Record<string, unknown>
+> {
+  const state = await getBackendState(req.errors, req.env);
+  const { id } = req.query as DeleteCustomTypeQuery;
+  const ctFolder = CustomTypesPaths(state.env.cwd).customType(id).folder();
+  const customTypeAssetsFolder = GeneratedCustomTypesPaths(state.env.cwd)
+    .customType(id)
+    .folder();
+
+  try {
+    IO.CustomType.deleteCustomType(ctFolder);
```

Comment:
this should be awaited, so that we don't try to do the rest if it fails.

Reply:
This function is already synchronous, so we don't need to await it right? All the other fs operations we do are the same.